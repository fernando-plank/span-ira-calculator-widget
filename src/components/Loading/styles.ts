import styled, { keyframes } from 'styled-components'

export const Wrapper = styled.div`
    width: 100vw;
    height: 100vh;
    background-color: #000;
    opacity: 0.7;
    position: fixed;
    left: 0;
    top: 0;
    z-index: 100;
`

const breatheAnimation = keyframes`
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
`

export const Loading = styled.div`
    display: inline-block;
    width: 80px;
    height: 80px;
    top: calc(50% - 40px);
    left: calc(50% - 40px);
    position: fixed;

    &:after {
        content: " ";
        display: block;
        width: 64px;
        height: 64px;
        margin: 8px;
        border-radius: 50%;
        border: 6px solid #fff;
        border-color: #fff transparent #fff transparent;
        animation: lds-dual-ring 1.2s linear infinite;
        animation-name: ${breatheAnimation}
    }
`;
