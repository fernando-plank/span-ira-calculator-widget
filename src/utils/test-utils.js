var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
