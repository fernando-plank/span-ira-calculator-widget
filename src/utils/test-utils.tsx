/* eslint-disable import/export */
import { ReactElement } from 'react'
import { RenderOptions, render } from '@testing-library/react'

import theme from '@styles/theme'
import { ThemeProvider } from 'styled-components'

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (ui: ReactElement, { ...options }: RenderOptions = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options)

export * from '@testing-library/react'
export { customRender as render }
