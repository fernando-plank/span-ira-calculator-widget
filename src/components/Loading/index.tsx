import Icon from '@components/Icon'
import * as S from './styles'
interface LoadingProps {
  isLoading: boolean
}

const Loading = (props: LoadingProps) => {
  return (
    props.isLoading && (
      <S.Wrapper>
        <S.LoadingCircle>
          <Icon icon={'AnimationCircle'}></Icon>
        </S.LoadingCircle>
        <S.LoadingPercentage>
          <Icon icon={'AnimationPercentage'}></Icon>
        </S.LoadingPercentage>
        <S.LoadingRotate>
          <Icon icon={'AnimationRotate'}></Icon>
        </S.LoadingRotate>
      </S.Wrapper>
    )
  )
}

export default Loading
