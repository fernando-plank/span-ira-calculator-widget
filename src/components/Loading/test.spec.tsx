import { screen } from '@testing-library/react'

import Loading from '.'
import { render } from '@utils/test-utils'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<Loading />', () => {
  it('should render the loading when isLoading is true', () => {
    const { container } = render(<Loading isLoading={true} />)

    expect(screen.getByTestId('loading')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should not render the loading when isLoading is true', () => {
    const { container } = render(<Loading isLoading={false} />)

    expect(screen.queryByTestId('loading')).not.toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
