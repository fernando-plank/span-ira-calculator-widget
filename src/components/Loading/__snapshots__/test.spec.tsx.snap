// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should not render the loading when isLoading is true 1`] = `null`;

exports[`<Loading /> should render the loading when isLoading is true 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  background-color: rgba(31,31,31,0.35);
  position: fixed;
  left: 0;
  top: 0;
  z-index: 100;
}

.c1 {
  top: calc(50% - 34px);
  left: calc(50% - 34px);
  position: fixed;
}

.c2 {
  top: calc(50% - 34px);
  left: calc(50% - 34px);
  position: fixed;
}

.c3 {
  top: calc(50% - 34px);
  left: calc(50% - 34px);
  position: fixed;
  z-index: 10;
}

.c3 > svg {
  -webkit-animation: lds-dual-ring 1.2s linear infinite;
  animation: lds-dual-ring 1.2s linear infinite;
  -webkit-animation-name: Hqxfx;
  animation-name: Hqxfx;
}

<div
  class="c0"
  data-testid="loading"
>
  <span
    class="c1"
  >
    <div>
      mocked icon
    </div>
  </span>
  <span
    class="c2"
  >
    <div>
      mocked icon
    </div>
  </span>
  <span
    class="c3"
  >
    <div>
      mocked icon
    </div>
  </span>
</div>
`;
